sequencer/src/state_signature/relay_server.rs

```diff
@@ -136,49 +133,41 @@ impl StateRelayServerDataSource for StateRelayServerState {
         }
         let block_height = state.block_height as u64;
         // TODO(Chengyu): this serialization should be removed once `LightClientState` implements `Eq`.
-        let bytes = to_bytes!(&state).unwrap();
-        let package = self.packages.entry(block_height).or_insert_with(|| {
+        let bundles_at_height = self.bundles.entry(block_height).or_insert_with(|| {
             self.queue.insert(block_height);
-            StateSignaturesPackage {
+            Default::default()
+        });
+        let bundle = bundles_at_height
+            .entry(state.clone())
+            .or_insert(StateSignaturesBundle {
                 state,
                 signatures: Default::default(),
                 accumulated_weight: U256::from(0),
-            }
-        });
-        let incoming_bytes = to_bytes!(&package.state).unwrap();
-        if bytes != incoming_bytes {
-            return Err(tide_disco::error::ServerError::catch_all(
-                StatusCode::BadRequest,
-                "The posted light client is not compatible with the one previously stored."
-                    .to_owned(),
-            ));
-        }
-        match package.signatures.entry(key) {
+            });
+        match bundle.signatures.entry(key) {
             std::collections::hash_map::Entry::Occupied(_) => {
-                return Err(tide_disco::error::ServerError::catch_all(
-                    StatusCode::BadRequest,
-                    "A signature is already posted at this block height for this key.".to_owned(),
-                ))
+                // A signature is already posted for this key with this state
+                return Ok(());
```

Comment:
In this case I think we could return an error or at least log something, since it is the client's own fault that they sent a duplicate signature

Reply:
Added back.
To further justify this, this error will trigger only if a **VALID** signature of this state is posted more than once. Could be some network issue, or your signing key is leaked.